src/cube_fit/linalg.py

```diff
@@ -166,3 +167,55 @@ def line_line_intersection(line1_point, line1_direction, line2_point, line2_dire
         return midpoint
 
     return intersection_point
+
+
+def nearest_orthogonal_matrix(l1, l2, l3):
+    M = np.vstack((l1,l2,l3)).T
+    U, S, V = np.linalg.svd(M)
+    R = U@V
+    return R
+
+
+def proj_point_on_plane(p0, n1, p1):
+    '''
+    p0 is a point to project on the plane
+    n1 is the plane's normal
+    p1 is a point on the plane
+    '''
+    # print("p0: ", p0, ", n1: ", n1, ", p1: ", p1)
+    p1p0 = p0 - p1
+    projp0 = p0 - (np.dot(p1p0, n1) / np.dot(n1, n1)) * n1
+    # print("p1p0: ", p1p0, ", p1projp0: ", (np.dot(p1p0, n1) / np.dot(n1, n1)) * n1, ", projp0: ", projp0)
+    return projp0
+
+
+def random_point_on_plane(n1, p1):
+    distance_limit = 1
+    random_coords = np.random.randn(3)
+    random_direction = random_coords - np.dot(random_coords, n1) / np.dot(n1, n1) * n1
+    random_direction /= np.linalg.norm(random_direction)
+    random_distance = np.random.uniform(0, distance_limit)
+    random_point = p1 + random_distance * random_direction
+    return random_point
+
+
+def distance_to_plane(p0,n1,p1):
```

Comment:
```suggestion
def distance_to_plane(p0, n1, p1):
```
Here too, you shouldn't have a local variable that has the same name as the function.